aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -451,33 +477,9 @@ Tensor quantized_index(const Tensor & self, const torch::List<c10::optional<Tens
   // For now, this is a naive implementation which does dq -> index -> q.
   // TODO(future PR): improve performance by removing the copies.
   const auto& self_dq = self.dequantize();
-
-  TORCH_CHECK_INDEX(indices.size() <= (size_t)self.dim(), "too many indices for tensor of dimension ", self.dim(), " (got ", indices.size(), ")");
-
-  auto info = make_info(self_dq, indices);
-  auto iter = make_index_iterator(info);
-  index_stub(iter.device_type(), iter, info.indexed_sizes, info.indexed_strides);
-  at::Tensor res = iter.output();
-
+  auto result = at::index(self_dq, indices);
```

Comment:
Not sure how important the extra overhead incurred from the dispatcher trip is here. We already know that we're in a QuantizedCPU kernel though, so maybe it's not too unreasonable to call `at::cpu::index`? If we do, we should rename the function `quantized_index` -> `quantized_cpu_index` to avoid confusion.

Reply:
Ah, you are right. That sounds way more reasonable.